File: 2015/02/solution1.rb
#!/usr/bin/env ruby

FILENAME = "puzzle_1_input.txt"

lines = File.readlines(FILENAME)

def calculate_sq_ft(l, w, h)
  sides = [
    l*w,
    w*h,
    h*l
  ]
  min = sides.min

  sides.reduce(0) { |sum, side| sum + 2*side } + min
end

total = 0

lines.each do |l|
  parts = l.split("x").map(&:to_i)
  feet = calculate_sq_ft(parts[0], parts[1], parts[2])
  total += feet
end

puts "answer: #{total}"

File: 2016/03/solution1.py
FILENAME = "input.txt"

if __name__ == "__main__":
  with open(FILENAME) as f:
    input = f.read()

  triangles = input.strip().split("\n")

  num_valid = 0
  for tri in triangles:
    sides = tri.split()
    sides = [int(side) for side in sides]
    sides.sort()

    if sides[0] + sides[1] > sides[2]:
      num_valid += 1

  print("answer: ", num_valid)
